src/Service.Tests/CosmosTests/SamplerTests.cs

```diff
@@ -180,23 +190,26 @@ public async Task TestGetPartitionInfoInPartitionBasedSampler()
         [TestMethod(displayName: "TimeBasedSampler Scenarios")]
         [DataRow(5, 1, 0, 5, DisplayName = "TimeBasedSampler: Retrieve 1 record, if it is allowed to fetch 1 item from a group and there are 5 groups (or time range)")]
         [DataRow(1, 10, 0, 10, DisplayName = "TimeBasedSampler: Retrieve 10 records, if it is allowed to fetch 10 item from a group and there is only 1 group.")]
-        [DataRow(null, 1, 0, 10, DisplayName = "TimeBasedSampler: Retrieve 2 records, if 1 item is allowed to fetch from each group and number of groups is 10 (i.e default)")]
+        [DataRow(null, 1, 0, 10, DisplayName = "TimeBasedSampler: Retrieve 10 records, if 1 item is allowed to fetch from each group and number of groups is 10 (i.e default)")]
         [DataRow(null, null, null, 10, DisplayName = "TimeBasedSampler: Retrieve records based on default values when no specific limits are set.")]
         [DataRow(5, 1, 4, 1, DisplayName = "TimeBasedSampler: Retrieve 1 record from a single group when records cannot be evenly divided into time-based groups.")]
         public async Task TestTimeBasedSampler(int? groupCount, int? numberOfRecordsPerGroup, int? maxDays, int expectedResultCount)
         {
             Mock<TimeBasedSampler> timeBasedSampler
                 = new(_containerWithNamePk, groupCount, numberOfRecordsPerGroup, maxDays, _mockLogger.Object);
-            if (maxDays is not null)
+
+            if (maxDays is null || maxDays == 0)
             {
-                timeBasedSampler
-                    .Setup<long>(x => x.GetTimeStampThreshold())
-                    .Returns((long)(_sortedTimespansNamePk[0] - maxDays));
+                maxDays = 10;
```

Comment:
same as previous comments: elaborate on why magic number 10

Reply:
resetting to default.